src/pages/messages/messages.ts

```diff
@@ -57,40 +59,135 @@ export class MessagesPage implements OnInit, OnDestroy {
   }
 
   ngOnInit() {
-    MeteorObservable.subscribe('messages', this.selectedChat._id).subscribe(() => {
-      MeteorObservable.autorun().subscribe(() => {
-        this.messages = Messages.find(
-          {chatId: this.selectedChat._id},
-          {sort: {createdAt: 1}}
-        ).map((messages: Message[]) => {
-          messages.forEach((message: Message) => {
-            message.ownership = this.senderId == message.senderId ? 'mine' : 'other';
-          });
-
-          return messages;
-        });
-      });
-    });
-
     this.autoScroller = this.autoScroll();
+    this.subscribeMessages();
+
+    // Get total messages count in database so we can have an indication of when to
+    // stop the auto-subscriber
+    MeteorObservable.call('countMessages').subscribe((messagesCount: number) => {
+      // Note that the 'scroller' element is being created dynamically by an Ionic
+      // component and therefore the event listener can't be registered directly
+      // from the view
+      let scrollListener = this.onScroll.bind(this);
+      this.scroller.addEventListener('scroll', scrollListener);
```

Comment:
Maybe you can wrap the scroll events with `Observable` and then treat the scroll changes as a stream.